libraries/AP_Math/AP_Math.cpp

```diff
@@ -129,54 +129,106 @@ float throttle_curve(float thr_mid, float alpha, float thr_in)
 }
 
 template <typename T>
-T wrap_180(const T angle, T unit_mod)
+T wrap_180(const T angle)
 {
-    auto res = wrap_360(angle, unit_mod);
-    if (res > T(180) * unit_mod) {
-        res -= T(360) * unit_mod;
+    auto res = wrap_360(angle);
+    if (res > T(180)) {
+        res -= T(360);
+    }
+    return res;
+}
+
+template <typename T>
+T wrap_180_cd(const T angle)
+{
+    auto res = wrap_360_cd(angle);
+    if (res > T(18000)) {
+        res -= T(36000);
+    }
+    return res;
+}
+
+template int wrap_180<int>(const int angle);
+template short wrap_180<short>(const short angle);
+template float wrap_180<float>(const float angle);
+#ifdef ALLOW_DOUBLE_MATH_FUNCTIONS
+template double wrap_180<double>(const double angle);
+#endif
+
+template int wrap_180_cd<int>(const int angle);
+template long wrap_180_cd<long>(const long angle);
+template short wrap_180_cd<short>(const short angle);
+template float wrap_180_cd<float>(const float angle);
+#ifdef ALLOW_DOUBLE_MATH_FUNCTIONS
+template double wrap_180_cd<double>(const double angle);
+#endif
+
+float wrap_360(const float angle)
+{
+    float res = fmodf(angle, 360.0f);
+    if (res < 0) {
+        res += 360.0f;
     }
     return res;
 }
 
-template int wrap_180<int>(const int angle, int unit_mod);
-template short wrap_180<short>(const short angle, short unit_mod);
-template float wrap_180<float>(const float angle, float unit_mod);
 #ifdef ALLOW_DOUBLE_MATH_FUNCTIONS
-template double wrap_180<double>(const double angle, double unit_mod);
+double wrap_360(const double angle)
+{
+    double res = fmod(angle, 360.0);
```

Comment:
Shouldn't you explicitly cast 360.0 to double ?

Reply:
With that compiler option indicating singlet-precision-floats it might be a good idea.

But it really doesn't matter AFAICS?